assets/js/modules/tagmanager/components/common/WebContainerNameTextField.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WebContainerNameTextField component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME, FORM_SETUP } from '../../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms';
+import ContainerNameTextField from './ContainerNameTextField';
+const { useSelect, useDispatch } = Data;
+
+export default function WebContainerNameTextField() {
+	const containers = useSelect( ( select ) => {
+		const accountID = select( STORE_NAME ).getAccountID();
+		return select( STORE_NAME ).getWebContainers( accountID );
+	} );
+
+	const siteName = useSelect( ( select ) => select( CORE_SITE ).getSiteName() );
+	const isSecondaryAMP = useSelect( ( select ) => select( CORE_SITE ).isSecondaryAMP() );
+	const containerName = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'containerName' ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	useEffect( () => {
+		if ( ! containerName ) {
+			setValues( FORM_SETUP, { containerName: siteName || global.location.hostname } );
```

Comment:
These fallbacks to `location.hostname` are there for the rare cases that `select( CORE_SITE )` is unpopulated, to prevent errors, correct? I'm assuming we'd rarely run into this in normal use, so just curious really.

Reply:
@tofumatt yep, you are right. It replicates the logic from the `Tag_Manager.php` class.